src/Coordination/Changelog.cpp

```diff
@@ -321,6 +321,8 @@ void Changelog::readChangelogAndInitWriter(uint64_t last_commited_log_index, uin
     else
         start_to_read_from = 1;
 
+    SCOPE_EXIT({ initialized = true; });
```

Comment:
Honestly, I'm not 100% sure... It looks like now we can get exceptions during initialization and still mark ourselves initialized.

Reply:
you're right, I should add check for `initialized` to all public functions using file writer and set it to true when it's okay.